prisma/migrations/20240803171237_setup_schema/migration.sql

```diff
@@ -0,0 +1,270 @@
+-- CreateEnum
+CREATE TYPE "TransactionType" AS ENUM ('DEPOSIT', 'PURCHASE');
+
+-- CreateEnum
+CREATE TYPE "Role" AS ENUM ('USER', 'INSTRUCTOR', 'ADMIN');
+
+-- CreateEnum
+CREATE TYPE "DifficultyLevel" AS ENUM ('BEGINNER', 'INTERMEDIATE', 'ADVANCED');
+
+-- CreateEnum
+CREATE TYPE "NotificationType" AS ENUM ('ASSIGNMENT_SUBMISSION', 'COURSE_PURCHASE');
+
+-- CreateTable
+CREATE TABLE "User" (
+    "id" TEXT NOT NULL,
+    "email" TEXT NOT NULL,
+    "password" TEXT NOT NULL,
+    "phoneNumber" TEXT NOT NULL,
+    "campus" TEXT NOT NULL,
+    "role" "Role" NOT NULL DEFAULT 'USER',
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+    "balance" DOUBLE PRECISION NOT NULL DEFAULT 0.0,
+
+    CONSTRAINT "User_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "Course" (
+    "id" TEXT NOT NULL,
+    "title" TEXT NOT NULL,
+    "description" TEXT NOT NULL,
+    "syllabus" TEXT NOT NULL,
+    "instructorId" TEXT NOT NULL,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+    "difficulty" "DifficultyLevel" NOT NULL,
+    "averageRating" DOUBLE PRECISION DEFAULT 0.0,
+
+    CONSTRAINT "Course_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "CourseEnrollment" (
+    "id" TEXT NOT NULL,
+    "userId" TEXT NOT NULL,
+    "courseId" TEXT NOT NULL,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "CourseEnrollment_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "CourseProgress" (
+    "id" TEXT NOT NULL,
+    "enrollmentId" TEXT NOT NULL,
+    "progress" DOUBLE PRECISION NOT NULL DEFAULT 0.0,
+    "lastAccess" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
```

Comment:
Since you are using timestamp convention of past tense with suffix `-at` e.g. `createdAt`, `updatedAt`, etc. I think better to also do the same for last accessed.

```suggestion
    "lastAccessedAt " TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
```